components/match2/wikis/deadlock/match_group_input_custom.lua

```diff
@@ -0,0 +1,181 @@
+		match.winner = nil
+	end
+
+	MatchGroupInputUtil.getCommonTournamentVars(match)
+
+	match.stream = Streams.processStreams(match)
+	match.extradata = MatchFunctions.getExtraData(match)
+
+	match.games = games
+	match.opponents = opponents
+
+	return match
+end
+
+---@param match table
+---@param opponents table[]
+---@return table[]
+function MatchFunctions.extractMaps(match, opponents)
+	local maps = {}
+	for key, map in Table.iter.pairsByPrefix(match, 'map', {requireIndex = true}) do
+		local finishedInput = map.finished --[[@as string?]]
+		local winnerInput = map.winner --[[@as string?]]
+
+		map.map = nil
+		map.participants = MapFunctions.getParticipants(map, opponents)
+		map.extradata = MapFunctions.getExtraData(map)
+
+		map.finished = MatchGroupInputUtil.mapIsFinished(map)
+		local opponentInfo = Array.map(opponents, function(_, opponentIndex)
+			local score, status = MatchGroupInputUtil.computeOpponentScore({
+				walkover = map.walkover,
+				winner = map.winner,
+				opponentIndex = opponentIndex,
+				score = map['score' .. opponentIndex],
+			}, MapFunctions.calculateMapScore(map.winner, map.finished))
+			return {score = score, status = status}
+		end)
+
+		map.scores = Array.map(opponentInfo, Operator.property('score'))
+		if map.finished or MatchGroupInputUtil.isNotPlayed(map.winner, finishedInput) then
+			map.resulttype = MatchGroupInputUtil.getResultType(winnerInput, finishedInput, opponentInfo)
+			map.walkover = MatchGroupInputUtil.getWalkover(map.resulttype, opponentInfo)
+			map.winner = MatchGroupInputUtil.getWinner(map.resulttype, winnerInput, opponentInfo)
+		end
+
+		table.insert(maps, map)
+		match[key] = nil
+	end
+
+	return maps
+end
+
+CustomMatchGroupInput.processMap = FnUtil.identity
+
+---@param maps table[]
+---@return fun(opponentIndex: integer): integer
+function MatchFunctions.calculateMatchScore(maps)
+	return function(opponentIndex)
+		return MatchGroupInputUtil.computeMatchScoreFromMapWinners(maps, opponentIndex)
+	end
+end
+
+---@param match table
+---@return table
+function MatchFunctions.getExtraData(match)
+	return {
+		comment = match.comment,
+	}
+end
+
+---@param map table
+---@return table
+function MapFunctions.getExtraData(map)
+	local extraData = {
+		comment = map.comment,
+		team1side = map.team1side,
+		team2side = map.team2side,
+	}
+
+	return extraData
+end
+
+---@param map table
+---@param opponents table[]
+---@return table
+function MapFunctions.getParticipants(map, opponents)
+	local participants = {}
+	local getCharacterName = FnUtil.curry(MatchGroupInputUtil.getCharacterName, HeroNames)
+
+	for opponentIndex in ipairs(opponents) do
+		for _, hero, playerIndex in Table.iter.pairsByPrefix(map, 't' .. opponentIndex .. 'h', {requireIndex = true}) do
+			participants[opponentIndex .. '_' .. playerIndex] = {
+				character = getCharacterName(hero),
+			}
+		end
+	end
+
+	return participants
+end
```

Comment:
probably needs adjusting with #4645

Reply:
yes

Reply:
Maybe, maybe not, depends on how the input looks like. Might not be possible to properly attach due to lack of info